src/agent.rs

```diff
@@ -594,6 +596,56 @@ impl AgentBuilder {
         self.cookie_store = Some(cookie_store);
         self
     }
+
+    /// Add middleware handler to this agent.
+    ///
+    /// All requests made by the agent will use this middleware. Middleware is invoked
+    /// in the order they are added to the builder.
+    pub fn middleware(mut self, m: impl Middleware + Send + Sync + 'static) -> Self {
+        if self.middleware.is_none() {
+            self.middleware = Some(vec![]);
+        }
+        self.middleware.as_mut().unwrap().push(Arc::new(m));
```

Comment:
Changing middleware from `Option` to `Vec` as I propose above lets us remove this unwrap.